### PR TITLE
Spaceship Project by @Ade179 @becky449 @Bludidi

### PR DESCRIPTION
**Pull request by Ade179**
### pair project by @Ade179  @Becky449  @Bludidi 

- linters passed
- use of classes and es6
- images of rockets laid out using flex and flex-wrap for responsiveness
- rocket counter effectively displayed as generated from the Spaceship API
- successful test carried out on comment, reservation and item(rocket) counter using jest 
- babel was installed as a dependency
- and js dom was installed as a testing environment for jest
- a video link to a sneak peek into our project:https://drive.google.com/file/d/1aL8t7U3M51w-nsEtEES_LBvZ8awua8tV/view?usp=share_link

### CODE REVIEW

File: src/modules/Modals.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    this.date_end = date_end;
+  }
+}
+
+export class Launch {
+  likes;
+
+   reserves = [];
+
+  comments = [];
```
Comment (ElsonOtake) [inline src/modules/Modals.js:22]: - [OPTIONAL] I think it's better to keep on code just the necessary variables. You can keep these variables if your code needs them.

File: src/modules/InvolvementUI.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+export default class InvolvementUI {
+  static renderComments = async (launchId) => {
+    document.getElementById('reserve').style.display = 'none';
+    document.getElementById('comment').style.display = 'block';
+    hiddenLaunchId.value = launchId;
+    getDetails.innerHTML = ' ';
+    await InvolvementService.getComments(launchId)
+      .then((comments) => {
+        commentsCount.innerText = InvolvementService.getCommentsCount(comments);
```
Comment (ElsonOtake) [inline src/modules/InvolvementUI.js:29]: - [ ] I think it's better to create a specific function to calculate the comments counter. A counter function should look for specific DOM elements and make the counting based on what is actually displayed on the page. Don't forget these count functions need to be covered with unit tests using Jest. Don't forget the case when the count is zero.

File: src/modules/launchUI.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const mission = launchModal.querySelector('.mission');
+const pad = launchModal.querySelector('.pad');
+const location = launchModal.querySelector('.location');
+const btnClose = document.querySelector('.close');
+btnClose.addEventListener('click', () => LaunchUI.closeModal());
+
+export default class LaunchUI {
+  static renderLaunches = () => {
+    LaunchService.getLaunches().then((launches) => {
+      launchCount.innerText = LaunchService.getLauchCount(launches);
```
Comment (ElsonOtake) [inline src/modules/launchUI.js:20]: - [ ] I think it's better to create a specific function to calculate the launches counter. A counter function should look for specific DOM elements and make the counting based on what is actually displayed on the page. Don't forget these count functions need to be covered with unit tests using Jest. Don't forget the case when the count is zero.

File: src/index.html [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            <form>
+              <input id="txt-visitor" type="text" placeholder="Your name" />
+              <input id="txt-comment" type="text" placeholder="Your comments" />
+              <input id="launch-id" type="hidden" />
+              <input id="btn-save-comment" type="button" value="Comment" />
+            </form>
+          </div>
+        </div>
+        <div id="reserve">
+          <h2>Reservation (<span class="reserve-count"></span>)</h2>
```
Comment (ElsonOtake) [inline src/index.html:73]: - [ ] I think it's better to create a specific function to calculate the reserve counter. A counter function should look for specific DOM elements and make the counting based on what is actually displayed on the page. Don't forget these count functions need to be covered with unit tests using Jest. Don't forget the case when the count is zero.

File: src/test/counterComments.test.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import InvolvementService from '../modules/InvolvementService.js';
+
+describe('Testing comments counter', () => {
```
Comment (ElsonOtake) [inline src/test/counterComments.test.js:7]: - [ ] Please create a new test for the implemented counter. I'm sure you will do your best.

File: src/test/counterItem.test.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import LaunchService from '../modules/LaunchService.js';
+
+describe('Testing items counter', () => {
```
Comment (ElsonOtake) [inline src/test/counterItem.test.js:7]: - [ ] Please create a new test for the implemented counter. I'm sure you will do your best.

File: src/test/reservationCounter.test.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import InvolvementService from '../modules/InvolvementService.js';
+
+describe('Testing reservation counter', () => {
```
Comment (ElsonOtake) [inline src/test/reservationCounter.test.js:7]: - [ ] Please create a new test for the implemented counter. I'm sure you will do your best.

**Review by ElsonOtake — CHANGES_REQUESTED** [review]
STATUS: CHANGES REQUIRED 🟡 

Hi @Ade179 @Bludidi @Becky449 ,

Good job so far! 🎉 🎉  I love your work!! 🏆 🏆 🏆 
There are some issues that you still need to work on to go to the next project but you are almost there! 🥲 🥲 

woohoo-youre-almost

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] Please recreate the video including the interesting piece of code or something you built that you are very proud of, max length of five minutes.
- [OPTIONAL] Please follow the requirements and make the code review for each other during the project. It's an important knowledge that you have to learn during this Microverse program and evaluated by recruiters. As the reviews were not performed in this project I will consider to fix these optional. But highly recommended for your next projects.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights

-  No linter errors. :heavy_check_mark: 
-  Correct Git flow. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Perfect Project. :heavy_check_mark: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**